### PR TITLE
Remote data

### PR DESCRIPTION
**Pull request by ian-r-rose**
Continuing with #15 

This uses remote data for the weights. I have uploaded the current weights files to a public s3 bucket, so anybody who pip installs can run the project right away.

The defaults can be overridden by specifying an on-disk (or other S3 bucket) on the command line,

### CODE REVIEW

File: automated_walk_bike_counter/core/tracking/object_tracker.py [hunk 1]
```diff
@@ -286,6 +286,10 @@ def trackObjects(self, parser ):
             ##boxes, scores, labels = gpu_nms(pred_boxes, pred_scores, args.num_class, max_boxes=30, score_thresh=0.5, nms_thresh=0.5)
             bxs, scrs, lbls = gpu_nms(pred_boxes, pred_scores, args.num_class, max_boxes=30, score_thresh=0.5, nms_thresh=0.5)
 
+            # Set a default AWS region for the Tensorflow S3 adapter.
+            # TODO: figure out a more robust way of handling this, so
+            # that we can have multiple regions or different storage backends.
+            os.environ['AWS_REGION'] = os.environ.get('AWS_REGION', 'us-west-1')
```
Comment (ian-r-rose) [inline automated_walk_bike_counter/core/tracking/object_tracker.py]: This is a hack that I couldn't really find a way around. If somebody wants to use an S3 bucket in a different region, they would have to export an environment variable for `AWS_REGION` to override this. It shouldn't affect the usage of local checkpoint files.

File: automated_walk_bike_counter/gui/widgets/video_frame.py [hunk 2]
```diff
@@ -54,7 +54,8 @@ def initialize_canvas(self):
         self.making_loading_frames()
 
         self.detection_thread_event = Event()
-        self.object_detection_thread = threading.Thread(target=self.controller.update_video_canvas, args=('data/my_data/sample.mp4',self), daemon=True)
+        sample_video_path = os.path.join(os.path.dirname(__file__), "../static/videos/sample.mp4")
+        self.object_detection_thread = threading.Thread(target=self.controller.update_video_canvas, args=(sample_video_path, self), daemon=True)
```
Comment (vahedi25411) [inline automated_walk_bike_counter/gui/widgets/video_frame.py]: The sample video path should be removed from the list of arguments in this line and also from the parameters' list of update_video_canvas method. I tried to remove it but I have a few other errors that I could not fix them for now. So, let's commit this changes and I will make these changes in another commit later.

File: automated_walk_bike_counter/core/tracking/object_tracker.py (hunk 1 above)
Comment (ian-r-rose) [reply automated_walk_bike_counter/core/tracking/object_tracker.py]: The problem I ran into here is that tensorflow IO is not particularly customizable with regards loading data from S3, so it more-or-less has to be done with an environment variable:
https://github.com/tensorflow/examples/blob/master/community/en/docs/deploy/s3.md#configuration

I think you are right that we can grab that data from the `config.ini` and set the environment variable from that. Do you have any ideas about the best way to distribute a default config file?
Another option would be to manually cache the contents of the S3 bucket and then load it locally.

Let me know what you think the best way forward is, and I can take a crack at implementing.

File: automated_walk_bike_counter/gui/widgets/video_frame.py (hunk 2 above)
Comment (ian-r-rose) [reply automated_walk_bike_counter/gui/widgets/video_frame.py]: Okay, let me know when you have updated!

**Comment by ian-r-rose** [comment]
Thanks for the review @vahedi25411!

**Pull request merged**